src/app/login/login.component.spec.ts

```diff
@@ -1,26 +1,95 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
+import { FooterComponent } from '../footer/footer.component';
+import { Router, Routes } from '@angular/router';
+import { Location } from '@angular/common';
 
 import { LoginComponent } from './login.component';
+import { ProblemListComponent } from '../home/problem-list/problem-list.component';
+import { HomeComponent } from '../home/home.component';
+import { TutorialPageComponent } from '../home/tutorial-page/tutorial-page.component';
+import { PageNotFoundComponent } from '../page-not-found/page-not-found.component';
+import { ReferenceComponent } from '../reference/reference.component';
+import { IUserInfo } from './login.model';
+import { UserInfoService } from '../services/utility-services/user-info.service';
 
 describe('LoginComponent', () => {
   let component: LoginComponent;
   let fixture: ComponentFixture<LoginComponent>;
+  let location: Location;
+  let router: Router;
+  let userInfoService: UserInfoService;
+
+  const routes: Routes = [
+    { path: '', component: LoginComponent },
+    { path: 'reference', component: ReferenceComponent },
+    {
+      path: 'home',
+      component: HomeComponent,
+      children: [
+        { path: '', component: ProblemListComponent },
+        { path: 'problem', component: ProblemListComponent },
+        { path: 'tutorial', component: TutorialPageComponent }
+      ]
+    },
+    { path: 'notfound', component: PageNotFoundComponent },
+    { path: '**', redirectTo: '/notfound' }
+  ];
```

Comment:
Can we export the routes from somewhere and use it here, instead of keeping a replica in 2 places? Keeping it here means there will be a change to the test and the source code every time we change the routes